overlord/auth/auth.go

```diff
@@ -143,9 +144,21 @@ NextUser:
 	return nil, ErrInvalidAuth
 }
 
-// Authenticator returns MacaroonAuthenticator for current authenticated user represented by UserState
-func (us *UserState) Authenticator() *MacaroonAuthenticator {
-	return newMacaroonAuthenticator(us.StoreMacaroon, us.StoreDischarges)
+// Authenticator returns a store.Authenticator which uses the given
+// user's stored credentials, or nil if there are no credentials.
+func Authenticator(st *state.State, userID int) (store.Authenticator, error) {
+	var us *UserState
+	if userID != 0 {
+		var err error
+		us, err = User(st, userID)
```

Comment:
I had misunderstood the history of the "user" argument to each API handler -- I saw it was used in just one or two places, so gathered it was on the way out, rather than being a recent addition that wasn't yet fully integrated. I've reworked daemon.auther() to take a user ID from the user argument instead of reparsing request, which simplifies things considerably.

Do you feel strongly about the auth.User(st, userID) call inside Authenticator? Authenticator needs to read from state anyway to get device credentials, and it seems quite weird to have it take both a state.State and an auth.UserState.